src/fu-tool.c

```diff
@@ -3067,6 +3073,48 @@ fu_util_switch_branch(FuUtilPrivate *priv, gchar **values, GError **error)
 	return fu_util_prompt_complete(priv->completion_flags, TRUE, error);
 }
 
+static gboolean
+fu_util_setup_console(GError **error)
+{
+#ifdef _WIN32
+	HANDLE hOut;
+	DWORD dwMode = 0;
+
+	/* workaround Windows setting the codepage to 1252 */
+	g_setenv("LANG", "C.UTF-8", FALSE);
+
+	/* enable VT sequences */
+	hOut = GetStdHandle(STD_OUTPUT_HANDLE);
+	if (hOut == INVALID_HANDLE_VALUE) {
+		g_set_error(error,
+			    G_IO_ERROR,
+			    G_IO_ERROR_NOT_SUPPORTED,
+			    "failed to get stdout [%u]",
+			    (guint)GetLastError());
+		return FALSE;
+	}
+	if (!GetConsoleMode(hOut, &dwMode)) {
+		g_set_error(error,
+			    G_IO_ERROR,
+			    G_IO_ERROR_NOT_SUPPORTED,
+			    "failed to get mode [%u]",
+			    (guint)GetLastError());
+		return FALSE;
+	}
+	dwMode |= ENABLE_VIRTUAL_TERMINAL_PROCESSING;
+	if (!SetConsoleMode(hOut, dwMode)) {
+		g_set_error(error,
+			    G_IO_ERROR,
+			    G_IO_ERROR_NOT_SUPPORTED,
+			    "failed to set mode [%u]",
+			    (guint)GetLastError());
+		return FALSE;
+	}
```

Comment:
This make Windows pretty?

Reply:
Gahh, ETOOMANYBRANCHES. Maybe. Will try in wip/hughsie/cros-ec-progress

Reply:
Well, it makes the text multicoloured now. There's still some UTF-8 issue.